parsers/NTESMO.py

```diff
@@ -112,9 +117,15 @@ def get_data(
     extraction_func: Callable[[bytes], pd.DataFrame],
     logger: Logger,
 ) -> pd.DataFrame:
+
+    # Get the default datetime if the target datetime is None
+    target_datetime = get_default_datetime_if_not_None(target_datetime)
```

Comment:
Same, the datetime cannot be None here according to the type hints.
So rather put this one level above in the fetch functions.

Reply:
Thanks for your comments! Instead of moving this into the fetch functions, would it be better to change the type hint to `target_datetime: Optional[datetime] = None`? I think this is a good fix, and similar to what appears in other parsers (e.g., [NL](https://github.com/electricitymaps/electricitymaps-contrib/blob/master/parsers/NL.py) - see the `fetch_production` definition).